docs/guides/code-samples-without-github-actions.md

```diff
@@ -0,0 +1,98 @@
+# Using the CLI to populate code samples without GitHub Actions
+
+This guide explains how to use the Speakeasy CLI to generate and publish code samples for your API documentation when you're not using GitHub Actions.
+
+## Overview
+
+Speakeasy can automatically generate code samples for your OpenAPI specification and make them available via a public URL that can be integrated with documentation platforms like Scalar. While this process is typically automated with GitHub Actions, you can achieve the same result using just the CLI in your own custom workflows.
+
+## Prerequisites
+
+- Speakeasy CLI installed
+- An OpenAPI specification
+- A Speakeasy account and API key
+
+## Step-by-step process
+
+### 1. Configure your workflow
+
+First, set up your workflow configuration file:
+
+```bash
+speakeasy configure
+```
+
+This creates a `.speakeasy/workflow.yaml` file that defines your SDK generation targets and code samples configuration.
+
+### 2. Generate SDKs and code samples
+
+Run the Speakeasy CLI to generate your SDKs and code samples:
+
+```bash
+speakeasy run
+```
+
+This command:
+- Downloads or loads your OpenAPI document
+- Validates the OpenAPI document
+- Generates SDKs for your configured languages
+- Creates code samples for each operation in your API
+
+### 3. Promote code samples to main
+
+The critical step for enabling automated code samples is to tag your generated code samples with the `main` tag:
+
+```bash
+speakeasy tag promote -c my-target-name -t main
+```
+
+This command marks your code samples as "official" so they can be incorporated into the public URL. Replace `my-target-name` with the name of your target as defined in your workflow configuration.
```

Comment:
For this section, be more clear about why promoting to `main` is important. Maybe draw a parallel to github, since the `main` branch in github is significant in that it usually represents the most production-ready code.